test/functional/cmdLineTests/jvmtitests/src/com/ibm/jvmti/tests/fieldwatch/fw001.java

```diff
@@ -0,0 +1,162 @@
+	public static native int modifyWatches(String className, String fieldName, String fieldSignature, boolean isStatic, boolean isRead, boolean isWrite, boolean isAdd);
+
+	public static int incrementNumReports()
+	{
+		return numReports.incrementAndGet();
+	}
+
+	public static int getNumReports()
+	{
+		return numReports.get();
+	}
+
+	public static void setNumReports(int i)
+	{
+		numReports.set(i);
+	}
+
+	public String helpFieldWatch()
+	{
+		return "Fieldwatch test 1";
+	}
+
+	public boolean setup(String args)
+	{
+		setNumReports(0);
+		return startTest(this.getClass().getName().replace('.', '/')) == JNI_OK ? true : false;
+	}
+
+	public boolean teardown()
+	{
+		return endTest() == JNI_OK ? true : false;
+	}
+
+	private boolean assertEquals(int expected, int actual)
+	{
+		if (expected != actual) {
+			System.err.println("Expected: " + expected + ", but got: " + actual);
+
+			try {
+				throw new Exception("Test failure due to unexpected value.");
+			} 
+			catch (Exception e) {
+				e.printStackTrace();
+			}
+
+			success = false;
+		}
+
+		return success;
+	}
+
+	public boolean testFieldWatch() throws InterruptedException
+	{
+		success = true;
+
+		System.out.println("=============== Beginning - STATIC AND INSTANCE FIELD WATCH TESTS\n\n");
+
+		boolean isRead = true;
+		boolean isStatic = true;
+		MyObject myObj = new MyObject();
+
+		// Add watches and generate reports for static field reads/writes.
+		runTest(myObj, isStatic, isRead);
+
+		runTest(myObj, isStatic, !isRead);
+
+		// Add watches and generate reports for instance field reads/writes.
+		runTest(myObj, !isStatic, isRead);
+
+		setNumReports(0);
+		runTest(myObj, !isStatic, !isRead);
+
+		System.out.println("=============== Ending - STATIC AND INSTANCE FIELD WATCH TESTS\n\n");
+		return success;
+	}
+
+	public void runTest(MyObject myObj, boolean isStatic, boolean isRead) throws InterruptedException
+	{
+		// The fields that we will be adding watches for all start with the following prefix.
+		String prefix = isStatic ? "static" : "instance";
+		// List containing suffix of each field name and its signature.
+		String[][] fieldData = {{"DoubleField","D"}, {"IntField","I"}, {"ObjField", "Ljava/lang/Object;"}, {"SingleField","F"}, {"LongField", "J"}};
+
+		for (String[] currRow : fieldData)
+		{
+			// Reset number of reported events.
+			setNumReports(0);
+
+			System.out.println("===============adding " + (isRead ? "field read" : "field write") + " watches for " + prefix + currRow[0] + "\n\n");
+
+			// Add field watch.
+			if (!assertEquals(JNI_OK, modifyWatches(myObj.getClass().getName().replace('.', '/'), prefix + currRow[0], currRow[1], isStatic, isRead, !isRead /*isWrite*/, true /* isAdd */))) {
+				System.out.println("*ERROR*: Unable to add field watches for " + prefix + currRow[0] + "\n");
+			}
+			System.out.println("===============run\n\n");
+
+			// Call Test routines that will invoke the watch events.
+			for (int j = 0; j < numExpectedReports; j++)
+			{
```

Comment:
Belongs on previous line.

Reply:
This and all other such issues are fixed in 255c1d6